### ISSUE TITLE
feature to resolve issue with shaders being unable to read nameplates and related

### ISSUE THREAD

**Issue by Juesto** [sp614x/optifine#3128]
"dedicated shader programs for chunk viewer lines, fishing lines, entity shadows, and entity nameplates would be nice, or, alternatively, make all "lines" use gbuffers_basic"
additional programs for entity nameplates, entity shadows, and debug lines would be very useful for shaders.
"the whole issue is that weird things like that don't have the same vertex format as regular geometry. for example, it's missing a normal vector, texture coordinates, and a few attributes that optifine normally provides. because of this, shaders get garbage data when they try to use those attributes. there's no way for the shader to tell if the attributes actually exist in the first place."

tldr: provide a separate program for the rest of 2D objects in 3D space just like block outlines already do have.

credit to Builderb0y for the explanation.
